internal/provider/user_resource_test.go

```diff
@@ -0,0 +1,140 @@
+package provider
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccUserResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read testing
+			{
+				Config: providerConfig + `
+					resource "trocco_user" "test" {
+					  email = "test@example.com"
+					  password = "3XRambMkp-Hw"
+					  role = "admin"
+					}
+				`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("trocco_user.test", "email", "test@example.com"),
+					resource.TestCheckResourceAttr("trocco_user.test", "role", "admin"),
+					resource.TestCheckResourceAttr("trocco_user.test", "can_use_audit_log", "false"),
+					resource.TestCheckResourceAttr("trocco_user.test", "is_restricted_connection_modify", "false"),
+					resource.TestCheckResourceAttrSet("trocco_user.test", "id"),
+				),
+			},
+			// ImportState testing
+			{
+				ResourceName:            "trocco_user.test",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"password"},
+			},
+			// Update and Read testing
+			{
+				Config: providerConfig + `
+					resource "trocco_user" "test" {
+					  email = "test@example.com"
+					  role = "member"
+					  can_use_audit_log = true
+					  is_restricted_connection_modify = true
+					}
+				`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("trocco_user.test", "email", "test@example.com"),
+					resource.TestCheckResourceAttr("trocco_user.test", "role", "member"),
+					resource.TestCheckResourceAttr("trocco_user.test", "can_use_audit_log", "true"),
+					resource.TestCheckResourceAttr("trocco_user.test", "is_restricted_connection_modify", "true"),
+				),
+			},
+		},
+	})
+}
```

Comment:
ASK: The acceptance tests for users fail in my environment. Could you tell me if something is wrong?

```
$ TROCCO_TEST_URL=https://... TROCCO_API_KEY=... make testacc

=== RUN   TestAccUserResource
    user_resource_test.go:11: Step 1/3 error: Error running apply: exit status 1
        
        Error: Creating user
        
          with trocco_user.test,
          on terraform_plugin_test.tf line 16, in resource "trocco_user" "test":
          16:                                   resource "trocco_user" "test" {
        
        Unable to create user, got error: 422 Unprocessable Content
```

Reply:
Your password strength might be too low. Please check this document.
https://documents.trocco.io/v2/docs/ja/password-policy